tests/test_protocols/test_abci/helper.py

```diff
@@ -0,0 +1,667 @@
+    Request,
+    Response,
+)
+from packages.valory.connections.abci.tendermint.crypto import (  # type: ignore
+    keys_pb2,
+    proof_pb2,
+)
+from packages.valory.connections.abci.tendermint.types import (  # type: ignore
+    params_pb2,
+    types_pb2,
+)
+from packages.valory.protocols import abci as valory_abci_protocol
+from packages.valory.protocols.abci import AbciMessage
+from packages.valory.skills.abstract_round_abci.dialogues import AbciDialogues
+
+
+Node = Dict[str, Any]
+MODULE_PATH = Path(*tendermint.__package__.split(".")).absolute()
+python_files = tuple(MODULE_PATH.rglob("*/*.py"))
+keys = ["_".join(f.parts[-2:])[:-3] for f in python_files]
+
+
+def _exec_module_import(path: str) -> ModuleType:
+    """Execute the import of a module"""
+    spec = importlib.util.spec_from_file_location(path, path)
+    module = importlib.util.module_from_spec(spec)  # type: ignore
+    spec.loader.exec_module(module)  # type: ignore
+    return module
+
+
+def get_aea_type(data_type: str) -> str:
+    """Translate type to AEA-native ABCI format"""
+    if data_type in type_to_python:
+        if type_to_python[data_type] is None:
+            raise NotImplementedError(f"type_to_python: {data_type}")
+        return f"pt:{type_to_python[data_type].__name__}"
+    return f"ct:{data_type}"
+
+
+type_mapping = {
+    v: k[5:].lower() for k, v in vars(FieldDescriptor).items() if k.startswith("TYPE_")
+}
+
+type_to_python = dict.fromkeys(type_mapping.values())
+type_to_python.update(
+    dict(
+        double=float,
+        float=float,
+        int64=int,
+        uint64=int,
+        int32=int,
+        bool=bool,
+        string=str,
+        bytes=bytes,
+        uint32=int,
+    )
+)
+
+
+camel_to_snake = _camel_case_to_snake_case  # ???
+snake_to_camel = _to_camel_case
+
+
+TENDERMINT_PRIMITIVES = tuple(type_mapping.values())
+PYTHON_PRIMITIVES = (int, float, bool, str, bytes)
+
+# simple AEA strategy
+NON_DEFAULT_PRIMITIVES = {str: "sss", bytes: b"bbb", int: 123, float: 3.14, bool: True}
+REPEATED_FIELD_SIZE = 3
+USE_NON_ZERO_ENUM: bool = True
+
+# simple Tendermint strategy
+SIMPLE_STRATEGY = dict(  # STRATEGY_MAP
+    int32=-32,
+    uint32=32,
+    int64=-64,
+    uint64=64,
+    bool=True,
+    string="sss",
+    bytes=b"bbb",
+    nanos=999,
+)
+
+
+MODULES = {}
+for name, file in zip(keys, python_files):
+    MODULES[name] = _exec_module_import(str(file))
+
+
+def is_enum(d_type: Any) -> bool:
+    """Check if a type is an Enum (not instance!)."""
+    return isinstance(d_type, type) and issubclass(d_type, Enum)
+
+
+def my_repr(self: Any) -> str:
+    """Custom __repr__ for Tendermint protobuf objects, which lack it."""
+    return f"<{self.__module__}.{type(self).__name__} object at {hex(id(self))}>"
+
+
+def is_repeated(d_type: Any) -> bool:
+    """Check if field is repeated."""
+    return d_type.__class__.__module__ == "typing" and d_type.__origin__ is list
```

Comment:
A bit hacky and might not work for all versions of python? Also, the name could be more descriptive, e.g., `is_typing_list`.

Reply:
this will not work if there is no annotation, for example, or when the annotation is wrong. In a non-hardcoded language I don't see a way around this, unless we start adding additional information elsewhere.

Do you have any suggestions with regard to your comment (outside renaming)? Note that everything in python is an object so `.__class__.__name__` is always accessible and `__origin__` keeps a reference the subscripted type